setup.py

```diff
@@ -152,7 +152,15 @@ class Extension(extension.Extension, object):
 
 python_requires = '>=3.7'
 install_requires = [
-    "scipy", "networkx", "dwave-networkx>=0.8.10", "numpy", "fasteners", "homebase", "rectangle-packer>=2.0.1"
+    "numpy==1.22.3;python_version>=3.8",
```

Comment:
You want to make the versions in `install_requires` as loose as possible. You also don't need to switch based on Python version, `pip` should handle that.